provider/server/machine_delete_test.go

```diff
@@ -80,7 +181,29 @@ var _ = Describe("DeleteMachine", func() {
 		})
 		Expect(err).NotTo(HaveOccurred())
 
-		By("ensuring machine is deleted")
+		By("ensuring machine is in Terminating state after delete")
+		Eventually(func() iri.MachineState {
+			listResp, err := machineClient.ListMachines(ctx, &iri.ListMachinesRequest{
+				Filter: &iri.MachineFilter{
+					Id: createResp.Machine.Metadata.Id,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(listResp.Machines).Should(HaveLen(1))
+			return listResp.Machines[0].Status.State
+		}).Should(Equal(iri.MachineState_MACHINE_TERMINATING))
+
+		By("ensuring machine is not deleted until gracefulShutdownTimeout")
+		time.Sleep(gracefulShutdownTimeout)
```

Comment:
Here, instead of using time.Sleep() to wait for the machine not to be deleted until gracefulShutdownTimeout, we can improve the test by checking that the machine exists consistently over that interval. This approach ensures that we continuously verify the machine's existence instead of waiting passively.

Reply:
@so-sahu I had an offline discussion with @lukas016 on this and decided to go ahead with time.sleep() for now.

Reply:
it doesn't make sense check it in loop, machine has to still exist after graceful period in this test.

Reply:
As discussed with @lukas016 offline, the sleep currently ensures that the machine is not deleted until the gracefulShutdownTimeout. Instead of passively assuming that the machine still exists, it would be beneficial to actively check for its existence during the graceful timeout period using a `consistently` block.